File: Samples/Translator-API/translation-nmt.py
import os
import requests
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Configuration
ENDPOINT = "https://api.cognitive.microsofttranslator.com/translate"
SUBSCRIPTION_KEY = os.getenv("SUBSCRIPTION_KEY")
REGION = os.getenv("REGION")

def translate_text(text, from_lang, to_lang):
    headers = {
        "Ocp-Apim-Subscription-Key": SUBSCRIPTION_KEY,
        "Ocp-Apim-Subscription-Region": REGION,
        "Content-Type": "application/json",
    }
    
    params = {"api-version": "2025-05-01-preview"}

    body = [{
        "Text": text,
        "language": from_lang,
        "targets": [{
            "language": to_lang
        }]
    }]
    
    response = requests.post(ENDPOINT, params=params, headers=headers, json=body, timeout=30)
    response.raise_for_status()
    
    data = response.json()
    return data[0]["translations"][0]["text"]

if __name__ == "__main__":
    translated_text = translate_text("J'ai un problème avec ma commande.", "fr", "en")
    print(translated_text)

File: Samples/Translator-API/TypeScript/translation-nmt.ts
import axios, { AxiosError } from 'axios';
import 'dotenv/config';

const globalEndpoint = 'https://api.cognitive.microsofttranslator.com';

const apiKey = process.env.API_KEY;
const region = process.env.REGION;

type TranslationResult = {
    detectedLanguage?: { language: string; score: number };
    translations: Array<{ text: string; to?: string; language?: string }>;
}

async function translateText(text: string, from: string, to: string): Promise<TranslationResult[]> {

    const headers: Record<string, string> = {
        'Ocp-Apim-Subscription-Key': apiKey!,
        'Ocp-Apim-Subscription-Region': region!,
        'Content-type': 'application/json',
    };

    const params = new URLSearchParams({ 'api-version': '2025-05-01-preview' });
    
    const body = [ 
        {
            text,
            language: from,
            targets: [{language: to}],
        },
    ]

    const response = await axios.post<TranslationResult[]>(`${globalEndpoint}/translate`, body, { headers, params });
    return response.data;
}

export async function main(): Promise<void> {
    const textToTranslate = "J'ai un problème avec ma commande.";
    const fromLanguage = 'fr';
    const targetLanguage = 'en';

    const translations = await translateText(textToTranslate, fromLanguage, targetLanguage);
    
    console.log('Translation Results:');
    console.log(`- [${targetLanguage}]: ${translations[0].translations[0].text}`);
}

main().catch((err) => {
    console.error('The sample encountered an error:', err instanceof Error ? err.message : err);
    process.exit(1);
});